src/components/implementation/no_interface/scheddev/sl_plugins.h

```diff
@@ -48,4 +48,8 @@ void sl_mod_init(void);
 void sl_timeout_mod_expended(cycles_t now, cycles_t oldtimeout);
 void sl_timeout_mod_init(void);
 
+/* API for thread wakeup using timeout module */
```

Comment:
Yes, you're right. I do not have oneshot timeout requests exposed. 
Can I just extend `sl_thd_block()` to include a parameter for `timeout`? The Timeout API for oneshot `block` requests is `void sl_timeout_mod_block(struct sl_thd *t, cycles_t wakeup);` is already implemented. But I think I need to expose it through `sl_thd_XX` API.